### PR TITLE
Named page 23682

### PR DESCRIPTION
**Pull request by dletorey**
### Description

Documenting the named page property for @page rule and added page property.

### Motivation

Working on the Named page feature of the @page rule 

### Additional details

Updated the existing example too

### Related issues and pull requests

[#23682 ](https://github.com/mdn/content/issues/23682)
[Related PR in BCD](https://github.com/mdn/browser-compat-data/pull/18922)

### CODE REVIEW

File: files/en-us/web/css/@page/page/index.md [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+page-type: css-at-rule-property
+tags:
+  - "@page"
+  - At-rule property
+  - CSS
+  - CSS Property
+  - NeedsBrowserCompatibility
+  - Reference
+  - Web
+browser-compat: css.at-rules.page.page
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:13]:
```suggestion
browser-compat: css.properties.page
```

I don't know if this exists.

File: files/en-us/web/css/@page/page/index.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
+---
+title: page
+slug: Web/CSS/@page/page
+page-type: css-at-rule-property
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:4]: There are `css-at-rule-descriptor` and `css-property` page types, but no `css-at-rule-property` page types. This is regular CSS property, associated with the @page, but not a descriptor. We need to move the page and have it listed as:

 ```suggestion
title: page
slug: Web/CSS/page
page-type: css-property
```

File: files/en-us/web/css/@page/page/index.md [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+  - CSS Property
+  - NeedsBrowserCompatibility
+  - Reference
+  - Web
+browser-compat: css.at-rules.page.page
+---
+
+{{CSSRef}}
+
+The **`page`** [CSS](/en-US/docs/Web/CSS) [at-rule](/en-US/docs/Web/CSS/At-rule) property, used with the {{cssxref("@page")}} at-rule. It is used to use a particular type of page called a **named page** that has been declared on a `@page` at-rule.
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:18]:
```suggestion
The **`page`** [CSS](/en-US/docs/Web/CSS) property is used to specify the named page, a specific type of page defined by the {{cssxref("@page")}}  [at-rule](/en-US/docs/Web/CSS/At-rule).
```

File: files/en-us/web/css/@page/page/index.md [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+If there are multiple selectors that are using a named page consecutively then a forced page break using [`break-after`](/en-US/docs/Web/CSS/break-after) is needed.
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:20]: is it required, or should we write: 
```suggestion
If there are multiple selectors that are using a named page consecutively then a forced page break using [`break-after`](/en-US/docs/Web/CSS/break-after) may be needed.
```

File: files/en-us/web/css/@page/page/index.md [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+page: auto;
+/* named page value using <custom-ident> */
+page: exampleName;
+```
+
+### Values
+
+- `auto`
+  - : Default setting for the property.
+- `name`
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:35]:
```suggestion
- {{cssxref("<custom-ident>")}}
```

File: files/en-us/web/css/@page/page/index.md [hunk 6]
```diff
@@ -0,0 +1,275 @@
...
+
+If there are multiple selectors that are using a named page consecutively then a forced page break using [`break-after`](/en-US/docs/Web/CSS/break-after) is needed.
+
+## Syntax
+
+```css
+/* default values */
+page: auto;
+/* named page value using <custom-ident> */
+page: exampleName;
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:28]:
```suggestion
/* set a named page */
page: exampleName;
page: chapterIntro;

/* Use ancestors named page */
page: auto; /* default value */

/* Global values */
page: inherit;
page: initial;
page: revert;
page: revert-layer;
page: unset;
```

I assume the global values are supported...

File: files/en-us/web/css/@page/page/index.md [hunk 7]
```diff
@@ -0,0 +1,275 @@
...
+```
+
+### Values
+
+- `auto`
+  - : Default setting for the property.
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:34]:
```suggestion
  - : Default value.  Use the value of the nearest ancestor with a non-`auto` value. If no ancestor has a named page value set, the empty string.
```

File: files/en-us/web/css/@page/page/index.md [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+- `name`
+  - : A value defined by a user on that is applied on the [`@page`](/en-US/docs/Web/CSS/@page) at-rule, this is case-sensitive and follows the syntax of [`<custom-ident>`](/en-US/docs/Web/CSS/custom-ident).
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:36]:
```suggestion
  - : Case-sensitive name defined in a [`@page`](/en-US/docs/Web/CSS/@page) at-rule.
```

File: files/en-us/web/css/@page/page/index.md [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+
+> **Note:** The page property does not inherit. However, if the page value on an element is auto, then its used value is the value specified on its nearest ancestor with a non-auto value. When specified on the root element, the used value for auto is the empty string.
```
Comment (estelle) [inline files/en-us/web/css/@page/page/index.md:38]:
```suggestion
> **Note:** The `page` property does not inherit. However, if the `page` value on an element is `auto`, then its used value is the value specified on its nearest ancestor with a non-`auto` value. When specified on the root element, the used value for auto is the empty string.
```

Comment (estelle) [reply files/en-us/web/css/@page/page/index.md:38]: I added this content as a suggested edit for `auto`, so maybe needs rework or omission. But if important, the note is ok too.

**Review by estelle — COMMENTED** [review]
I only reviewed the top half.

File: files/en-us/web/css/@page/page/index.md (hunk 1 above)
Comment (teoli2003) [reply files/en-us/web/css/@page/page/index.md:13]: I don't think we are speaking about the CSS property here, but from an at-rule descriptor (or at-rule property, I think the terminology varies from one spec to another).

I may be mistaken, and this is worth clarifying.

File: files/en-us/web/css/@page/index.md [hunk 10]
```diff
@@ -284,13 +296,24 @@ The page-margin properties are the set of CSS properties can be set in any indiv
...
+
+An example of this can be found on the [`page`](/en-US/docs/Web/CSS/@page/page#examples) examples.
+
 ## Examples
 
-The example below prints out 5 different sections in `landscape` size, and adds a 20% margin to each of them in the print-out.
+### Using the size property to change the page orientation
 
-HTML:
+This example show how you can split the `section`s into individual pages in `landscape` format and each page has a 20% margin when printed.
```
[the remark this answers is not in the document]
Comment (dletorey) [reply files/en-us/web/css/@page/index.md:311]: Thank you

File: files/en-us/web/css/@page/page/index.md (hunk 1 above)
Comment (dletorey) [reply files/en-us/web/css/@page/page/index.md:13]: Yes @teoli2003 is correct this is a property specific to the page at-rule
https://drafts.csswg.org/css-page-3/#using-named-pages
The at-rule allows a name, which can then be referenced via the page property.
[Referenced in this BCD file](https://github.com/CodeRedDigital/browser-compat-data/blob/named-page-23682/css/at-rules/page.json#L47-L81)

File: files/en-us/web/css/@page/page/index.md (hunk 2 above)
Comment (dletorey) [reply files/en-us/web/css/@page/page/index.md:4]: This makes sense to me now thank you

File: files/en-us/web/css/@page/index.md [hunk 11]
```diff
@@ -52,6 +57,13 @@ The `@page` at-rule can be accessed via the CSS object model interface {{domxref
 
 > **Note:** The W3C is discussing how to handle viewport-related {{cssxref("&lt;length&gt;")}} units, `vh`, `vw`, `vmin`, and `vmax`. Meanwhile do not use them within a `@page` at-rule.
 
+### Properties
+
+The `@page` at-rule, allows the user to assign a name to the rule, which is then called in a selector using the `page` property.
+
+- [`page`](/en-US/docs/Web/CSS/@page/page)
+  - : Allows a selector to use a user defined **named page**
```
[the remark this answers is not in the document]
Comment (dletorey) [reply files/en-us/web/css/@page/index.md:65]: I added it as a property as it doesn't seem to be [listed as a descriptor](https://drafts.csswg.org/css-page-3/#property-index)

File: files/en-us/web/css/page/index.md [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+fieldset {
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  gap: 1rem;
+  width: fit-content;
+}
+```
+
+The first part of the CSS sets up the **named** pages, these include the size and orientaion and also some content to go in the `@top-middle` margin of the printed pages.
```
Comment (estelle) [inline files/en-us/web/css/page/index.md:158]:
```suggestion
The first part of the CSS sets up the **named** pages, these include the size and orientation and also some content to go in the [`@top-middle` margin](/en-US/docs/Web/CSS/@page#margin_at-rules) of the printed pages.
```

File: files/en-us/web/css/page/index.md [hunk 13]
```diff
@@ -0,0 +1,277 @@
...
+
+@page chapter {
+  size: a4 landscape;
+  @top-middle {
+    content: "Chapter";
+  }
+}
+```
+
+The next part of the CSS specifies which CSS selectors will use which named pages when printing. As the sections with `class="chapter"` are concurrent they appear as one page, in order to split them up they are split using `break-after: page;` which splits each chapter into another page.
```
Comment (estelle) [inline files/en-us/web/css/page/index.md:197]:
```suggestion
The next part of the CSS uses [attribute selectors](/en-US/docs/Web/CSS/Attribute_selectors) to apply the print dimensions, orientation, and margins defined in the named  `@page` rules defined in the previous CSS section to elements using the `page` property. 

The sections with `class="chapter"` are concurrent and appear as one page. The `break-after: page;` is used to split them up, which splits each chapter into a separately printed page.
```

File: files/en-us/web/css/page/index.md [hunk 14]
```diff
@@ -0,0 +1,277 @@
...
+    break-after: page;
+  }
+  .chapter > ul {
+    columns: 2;
+  }
+}
+```
+
+#### JavaScript
+
```
Comment (estelle) [inline files/en-us/web/css/page/index.md:246]:
```suggestion

The JavaScript updates the value of the `data-print` attribute, which is the attribute on which the named page is applied, when you select a different  printing option:
```

File: files/en-us/web/css/page/index.md [hunk 15]
```diff
@@ -0,0 +1,277 @@
...
+  }
+});
+printButton.addEventListener("click", () => {
+  window.print();
+});
+```
+
+#### Result
+
+Selecting a different print style radio input will change the output of the print dialog.
```
Comment (estelle) [inline files/en-us/web/css/page/index.md:267]:
```suggestion
What is printed, and what is shown in the print preview dialog, will change depending on which print style radio button is selected.
```

**Review by estelle — COMMENTED** [review]
I made some suggestions.

File: files/en-us/web/css/@page/page/index.md (hunk 7 above)
Comment (dletorey) [reply files/en-us/web/css/@page/page/index.md]: Done

File: files/en-us/web/css/@page/page/index.md (hunk 9 above)
Comment (dletorey) [reply files/en-us/web/css/@page/page/index.md]: I updated the `auto` description and remooved this note.

File: files/en-us/web/css/page/index.md [hunk 16]
```diff
@@ -0,0 +1,278 @@
+---
+title: page
+slug: Web/CSS/page
+page-type: css-property
+browser-compat: css.at-rules.page.page
```
Comment (estelle) [inline files/en-us/web/css/page/index.md:5]:
```suggestion
browser-compat: css.properties.page
```

**Review by estelle — CHANGES_REQUESTED** [review]
The `browser-compat: css.at-rules.page.page` is incorrect. I think `css.properties.page` should work

File: files/en-us/web/css/page/index.md (hunk 16 above)
Comment (dletorey) [reply files/en-us/web/css/page/index.md:5]: Thank you for this @estelle I see now that there are 2 parts to this the `@page` `page` descriptor and the `page` property that uses the value of the descriptor.
Fixing now

Comment (dletorey) [reply files/en-us/web/css/page/index.md]: Done and updated the BCD too

**Review by estelle — APPROVED** [review]
Yay!

**Pull request merged**